src/vidpf.rs

```diff
@@ -385,21 +440,130 @@ impl From<VidpfServerId> for Choice {
 }
 
 /// Adjusts values of shares during the VIDPF evaluation.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 struct VidpfCorrectionWord<W: VidpfValue> {
     seed: VidpfSeed,
     left_control_bit: Choice,
     right_control_bit: Choice,
     weight: W,
 }
 
+impl<W: VidpfValue> PartialEq for VidpfCorrectionWord<W> {
+    fn eq(&self, other: &VidpfCorrectionWord<W>) -> bool {
+        let seed_and_weight_equal = if self.seed == other.seed && self.weight == other.weight {
+            1
+        } else {
+            0
+        };
+
+        0 == u8::conditional_select(
+            &seed_and_weight_equal,
+            &0,
+            self.left_control_bit
+                .ct_eq(&other.left_control_bit)
+                .bitand(self.right_control_bit.ct_eq(&other.right_control_bit)),
+        )
+    }
+}
+
+impl<W: VidpfValue> ParameterizedDecode<W::ValueParameter> for VidpfCorrectionWord<W> {
+    fn decode_with_param(
+        decoding_parameter: &W::ValueParameter,
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        let mut seed = [0; 16];
+        bytes.read_exact(&mut seed)?;
+        let control_bits = u8::decode(bytes)?;
+        let (left_control_bit, right_control_bit) = match control_bits {
+            0 => (Choice::from(0), Choice::from(0)),
+            1 => (Choice::from(0), Choice::from(1)),
+            2 => (Choice::from(1), Choice::from(0)),
+            3 => (Choice::from(1), Choice::from(1)),
+            _ => return Err(CodecError::UnexpectedValue),
+        };
+        let weight = W::decode_with_param(decoding_parameter, bytes)?;
+        Ok(Self {
+            seed,
+            left_control_bit,
+            right_control_bit,
+            weight,
+        })
+    }
+}
+
+impl<W: VidpfValue> Encode for VidpfCorrectionWord<W> {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        bytes.extend_from_slice(&self.seed);
+        let both_control_bits = if bool::from(self.left_control_bit) {
+            if bool::from(self.right_control_bit) {
+                3
+            } else {
+                2
+            }
+        } else if bool::from(self.right_control_bit) {
+            1
+        } else {
+            0
+        };
+        bytes.push(both_control_bits);
+        self.weight.encode(bytes)?;
+        Ok(())
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(17 + self.weight.encoded_len()?)
+    }
+}
```

Comment:
Following the lead of the `idpf` module, we would not encode/decode correction words individually. Rather, we would encode/decode the public share so that we can pack the control bits as tightly as possible. I think we're going to do the same thing here: we're trying to shave as many bytes off the wire as we can. 
https://github.com/divviup/libprio-rs/blob/967cd629aee9fe375332d4e635b6209bfb717a34/src/idpf.rs#L675-L679

I propose we punt this to the next PR, since we have plenty to review here.